src/main/java/com/epam/ta/reportportal/core/bts/handler/impl/GetTicketHandlerImpl.java

```diff
@@ -47,60 +47,94 @@
-		this.integrationRepository = integrationRepository;
+	public GetTicketHandlerImpl(PluginBox pluginBox, GetBugTrackingSystemHandler getBugTrackingSystemHandler,
+			ProjectRepository projectRepository) {
 		this.pluginBox = pluginBox;
+		this.getBugTrackingSystemHandler = getBugTrackingSystemHandler;
+		this.projectRepository = projectRepository;
 	}
 
 	@Override
-	public Ticket getTicket(String ticketId, Long integrationId, ReportPortalUser.ProjectDetails projectDetails) {
-		List<Integration> integrations = integrationRepository.findAllByProjectId(projectDetails.getProjectId());
-		expect(integrations, not(CollectionUtils::isEmpty)).verify(
-				PROJECT_NOT_CONFIGURED,
-				Suppliers.formattedSupplier("No integrations for project with id = '{}' were found", projectDetails.getProjectId())
-		);
-		Integration integration = integrations.stream()
-				.filter(it -> Objects.equals(it.getId(), integrationId))
-				.findFirst()
-				.orElseThrow(() -> new ReportPortalException(INTEGRATION_NOT_FOUND, integrationId));
+	public Ticket getTicket(String ticketId, String url, String btsProject, ReportPortalUser.ProjectDetails projectDetails) {
+
+		Project project = projectRepository.findById(projectDetails.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, projectDetails.getProjectName()));
+
+		Integration integration = getBugTrackingSystemHandler.getEnabledProjectIntegrationByUrlAndBtsProject(projectDetails,
+				url,
+				btsProject
+		).orElseGet(() -> {
+			Integration globalIntegration = getBugTrackingSystemHandler.getEnabledGlobalIntegrationByUrlAndBtsProject(url, btsProject)
+					.orElseThrow(() -> new ReportPortalException(ErrorType.INTEGRATION_NOT_FOUND, url));
+
+			IntegrationValidator.validateProjectLevelIntegrationConstraints(project, globalIntegration);
+
+			return globalIntegration;
+		});
 
 		Optional<BtsExtension> btsExtension = pluginBox.getInstance(integration.getType().getName(), BtsExtension.class);
 		expect(btsExtension, Optional::isPresent).verify(BAD_REQUEST_ERROR,
 				"BugTracking plugin for {} isn't installed",
 				integration.getType().getName()
 		);
-		return btsExtension.get().getTicket(ticketId, integration).get();
+
+		return btsExtension.get()
+				.getTicket(ticketId, integration)
+				.orElseThrow(() -> new ReportPortalException(ErrorType.TICKET_NOT_FOUND, ticketId));
 	}
 
 	@Override
 	public List<PostFormField> getSubmitTicketFields(String ticketType, Long integrationId,
 			ReportPortalUser.ProjectDetails projectDetails) {
-		Integration integration = validateExternalSystem(integrationId);
+
+		Project project = projectRepository.findById(projectDetails.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, projectDetails.getProjectName()));
+
+		Integration integration = getBugTrackingSystemHandler.getEnabledByProjectIdAndId(projectDetails, integrationId).orElseGet(() -> {
```

Comment:
i see the duplicates of getting integration logic in several places. CreateTicketHandler and down here too.
maybe we can move it somewhere? 
i think, that we can move it in getBugTrackingSystemHandler
create method getIntegartion(bla bla) and move this logic there

so, when we are reading code what we would see. getIntegration (okay, this method gets an integration)
if we need to understand logic we go inside and see that (getProjectIntegration or else getGlobal)
so, as i think that it is pretty simple to understand